### PR TITLE
orc: add version 2.0.3 (and other stable releases)

### PR DESCRIPTION
**Pull request by wgtmac**
### Summary
Changes to recipe:  **orc/2.0.3**

#### Motivation
Sync recent releases from Apache ORC.

#### Details
Add several news releases from Apache ORC 2.0.x, 1.9.x and 1.8.x major versions.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by wgtmac** [comment]
@toge @valgur Could you please take a look? Thanks!

**Comment by valgur** [comment]
+1 I would suggest keeping only the latest versions for each major release (1.9.5 and 2.0.3)..

**Comment by wgtmac** [comment]
Make sense! Now I just keep 1.9.5 and 2.0.3. Thanks @ErniGH and @valgur!

**Review by valgur — APPROVED** [review]

**Review by valgur — CHANGES_REQUESTED** [review]
Keep the old versions. :)

**Comment by wgtmac** [comment]
To be clear, do you mean that I need to keep 1.9.5, 2.0.0 and 2.0.3. Right? @valgur

**Comment by perseoGI** [comment]
@wgtmac  

> BTW, if someone is using 1.7.10 or 1.9.3. Does it break anything if these versions are removed?

It is safe to remove versions, they will be kept in CCI. The only difference is that those versions will stop receiving recipe revision updates.

**Review by ErniGH — APPROVED** [review]
LGTM

**Review by AbrilRBS — APPROVED** [review]

File: recipes/orc/all/conanfile.py [hunk 1]
```diff
@@ -114,11 +109,6 @@ def generate(self):
         tc.variables["STOP_BUILD_ON_WARNING"] = False
         tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
 
-        # CMake versions less than 3.12 are supported by ORC pre-1.9.0 versions.
-        # Setting policy CMP0077 to NEW to remove unnecessary cache_variables settings.
-        if self.version < "1.9.0":
```
[the remark this answers is not in the document]
Comment (jcar87) [reply recipes/orc/all/conanfile.py:119]: Are we sure about that? From what I can see it worked as intended
The versions in `conandata.yml` previous to this PR were: `2.0.0`, `1.9.3`, `1.9.2`, `1.8.7`, `1.8.6`, `1.7.10`
From what I can see, the pure string comparisons would have been correct for all these versions:

* `"2.0.0" < "1.9.0"`: `False`
* `"1.9.3" < "1.9.0"`: `False`
* `"1.9.2" < "1.9.0"`: `False`
* `"1.8.7" < "1.9.0"`: `True`
* `"1.8.6" < "1.9.0"`: `True`
* `"1.7.10" < "1.9.0"`: `True`

Lexicographical version comparisons are correct when the number of digits across versions is fixed, and the most significant components go from the left to right - 
There are potential issues when you have `1.2.0`, `1.19.0`, `1.20.0` - but they don't apply to this recipe and these versions - so surrounding with `Version()` is to account for that, especially if in the future versions like that are used (or prereleases, etc) - but in this case was not critical enough to be actually a bug.

**Pull request merged**